src/CommunityToolkit.Aspire.Hosting.Ollama/OllamaModelHealthCheck.cs

```diff
@@ -0,0 +1,25 @@
+using Aspire.Hosting.ApplicationModel;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using OllamaSharp;
+
+namespace CommunityToolkit.Aspire.Hosting.Ollama;
+
+internal class OllamaModelHealthCheck(OllamaModelResource modelResource) : IHealthCheck
+{
+    public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
+    {
+        (var success, var endpoint) = await OllamaUtilities.TryGetEndpointAsync(modelResource, cancellationToken);
+
+        if (!success || endpoint is null)
+        {
+            return HealthCheckResult.Unhealthy("Invalid connection string");
+        }
+
+        var ollamaClient = new OllamaApiClient(endpoint);
+
+        // this will only return once the model is downloaded
+        await ollamaClient.ShowModelAsync(modelResource.ModelName, cancellationToken);
```

Comment:
I feel like health checks are going to be inherently hard to test due to the complex nature of the state that needs to be managed for Ollama and the size of the various Docker images we're going to be relying on.

The models in our sample results in a 2.8gb volume, combine that with the 3.28gb and 4.2gb container images for Ollama and Open WebUI, it's the better part of 10gb to just run these tests as they stand. I _think_ this is part of the underlying problem as to why the tests for Ollama aren't working anymore and I'm hesitant to add more integration tests as a result.

Reply:
Ok, I've fixed the test hang as part of https://github.com/CommunityToolkit/Aspire/pull/203/commits/8da2350e82cca5e4d2949b1e7c4170d20015888a

I understand why the test was failing (that was the result of the refactor in the PR to move to using `KnownResourceStates` rather than our own custom `Ready` for the model), and I _think_ I understand why the timeout didn't trigger it to fail but I'm not sure it's something that is easily solvable in our project (nor should it be).

For doing the unhealthy and healthy checks, well, we do test for healthy indirectly by waiting for the resource to go to a `KnownResourceStates.Running`, but we could change that to `WaitForResourceHealthyAsync`.

With Unhealthy testing, we'd have to write a test that does `WaitForResourceAsync` with a callback looking at the `resourceEvent.Snapshot.HealthStatus` and then pass if we find `Unhealthy`. But this runs the risk of an order-of-execution problem, if it becomes healthy before it hits that event registration then we may never see the event and it'll cause the problem that I was just debugging (I _think_... I'm assuming that the events aren't buffered, which they may be).